src/draive/gemini/lmm_generation.py

```
@@ -275,287 +318,356 @@ async def _completion_stream(
+        match tools.selection:
+            case "auto":
+                function_calling_mode = FunctionCallingConfigMode.AUTO
+
+            case "required":
+                function_calling_mode = FunctionCallingConfigMode.ANY
+
+            case "none":
+                functions = None  # no need to pass functions if none can be used
+                function_calling_mode = FunctionCallingConfigMode.NONE
+
+            case _:  # TODO: handle specific tool selection?
```

Pylint (3.3.7)

[warning] TODO: handle specific tool selection?

(W0511)